### PR TITLE
1.0 Badge component

### LINKED ISSUE #150: Combine Badge and Pill
Rename to Pill

### PR DESCRIPTION
**Pull request by designmatty**
## Overview

- 1.0 Badge
- Small fix to Button (extra semi-colon)
- upgrades polished.js
- added `textTransform` to style-props

### Checklist

- [x] Relevant documentation pages have been created or updated
- [x] Description of PR is in an appropriate section of the changelog and grouped with similar changes if possible

### Upgrade instructions

If there are any of the following in this PR, provide proper instructions on how to upgrade:

- [ ] Changes to the prop names or types of existing components
- [ ] Changes in intended behavior of existing component props
- [ ] Changes in the theme file's structure
- [ ] A required version bump to a non-dev dependency of the project

### Demo

Optional. Screenshots, `curl` examples, etc.
image

Closes #150

### CODE REVIEW

File: packages/core/components/badge/index.mdx [hunk 1]
```diff
@@ -6,40 +6,148 @@ route: /components/badge
...
-| Prop | Type | Required | Default | Description |
-|:-----|:-----|:---------|:--------|:------------|
-| Text props | | | | see [Text](/typography/text) |
-| borderRadius | number or string | no | `small` | see [styled-system BORDERRADIUS](https://github.com/jxnblk/styled-system/blob/master/docs/api.md#borders) |
+### PropTypes
+| Prop | Type | Required | Default | Description | Themeable |
+|:-----|:-----|:---------|:--------|:------------|:----------|
+| appearance | enum | no | `default` | `["default", "prominent"]` |  |
+| intent | string | no | `default` | Theme provides styles for `["default", "secondary", primary", "success", "warning", "danger"]` | yes |
+| scale | string | no | | Theme provides styles for `["small", "large"` | yes |
```
Comment (lederer) [inline packages/core/components/badge/index.mdx]: Missing a closing `]` after `"large"`

**Review by lederer — APPROVED** [review]
Overall LGTM.

One concern: 3 of the intents result in insufficient color contrast with `default` appearance (`primary`: 4.48, `warning`: 3.61, `success`: 3.7). All intents are fine with `prominent` appearance. I don't know if you want to address this now, or consider it out-of-scope and address it with a theme-focused PR at some point.

**Comment by designmatty** [comment]
@lederer this is ready for another look.

**Comment by lederer** [comment]
Super neato. Contrast is fixed for all badges. 🎉

Couple of things:
1. The new contrast logic seems broadly applicable and perfect for extracting out into `utils.js` so it can be reused elsewhere.
2. `primary` and `secondary` badges are virtually indistinguishable with `default` appearance. They're more distinct with `prominent` appearance, though it wouldn't really be clear at a glance which one is more primary than the other. This may just come down to choosing different primary and secondary colors in the theme, and hence out of scope?

image

image

**Comment by designmatty** [comment]
1: I agree, but would want to make some changes to the logic in terms of how it handles the background. It's currently hardcoded but I'd want to be able to pass it as an argument. We can make an issue to do this later.

2: I've been considering the removal of `secondary` from the theme due to this. But I'd say out of scope for this PR.

**Comment by lederer** [comment]
👍 SGTM

**Pull request merged**